### PR TITLE
Issue #6916: migrate tests to junit5 for {Main,JavadocPropertiesenerator}Test

### PR DESCRIPTION
**Pull request by pbludov**
Issue #6916

Tests of the ` {Main,JavadocPropertiesenerator}Test` are migrated to Junit5.

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 1]
```diff
@@ -48,14 +50,16 @@
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.contrib.java.lang.system.ExpectedSystemExit;
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: can we remove https://github.com/checkstyle/checkstyle/blob/master/pom.xml#L292 dependency now ?

Comment (pbludov) [reply src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: Not yet. The dependency
```
    <!-- till https://github.com/checkstyle/checkstyle/issues/7368 -->
    <dependency>
      <groupId>com.github.stefanbirkner</groupId>
      <artifactId>system-rules</artifactId>
      <version>1.19.0</version>
      <scope>test</scope>
    </dependency>
```
is used in the Powermock tests.
Junit team is working on an issue to allow Powermock to replace the classloader.

File: src/test/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGeneratorTest.java [hunk 2]
```diff
@@ -78,73 +77,90 @@ protected final String getNonCompilablePath(String filename) throws IOException
...
+        systemOut.capture();
         JavadocPropertiesGenerator.main("--nonexistent-argument");
 
         final String expected = String.format(Locale.ROOT, "Missing required options "
                 + "[--destfile=<outputFile>, params[0]=<inputFile>]%n")
                 + USAGE;
-        assertEquals("Unexpected error log", expected, systemErr.getLog());
-        assertEquals("Unexpected output log", "", systemOut.getLog());
+        assertEquals(expected, systemErr.getCapturedData(), "Unexpected error log");
+        assertEquals("", systemOut.getCapturedData(), "Unexpected output log");
```
Comment (strkkk) [inline src/test/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGeneratorTest.java]: `assertEquals("", systemOut.getCapturedData()`

May be change to `assertTrue(systemOut.getCapturedData().isEmpty())` here and below?

File: src/test/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGeneratorTest.java [hunk 3]
```diff
@@ -78,73 +77,90 @@ protected final String getNonCompilablePath(String filename) throws IOException
...
+        assertTrue(TestUtil.isUtilsClassHasPrivateConstructor(
+            JavadocPropertiesGenerator.class, false), "Constructor is not private");
     }
 
     @Test
-    public void testNonExistentArgument() throws Exception {
+    public void testNonExistentArgument(@SysErr Capturable systemErr, @SysOut Capturable systemOut)
+            throws Exception {
+        systemErr.capture();
+        systemOut.capture();
```
Comment (strkkk) [inline src/test/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGeneratorTest.java]:
```
systemErr.capture();
        systemOut.capture();
```
looks like these lines are duplicated many times.
May be extract them to some method with @BeforeEach annotation?

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 4]
```diff
@@ -404,72 +394,74 @@ public void testGetOutputStreamOptionsMethod() throws Exception {
...
+        final LocalizedMessage invalidPatternMessageMain = new LocalizedMessage(1,
+                "com.puppycrawl.tools.checkstyle.checks.naming.messages",
+                "name.invalidPattern", new String[] {"InputMain", "^[a-z0-9]*$"},
+                null, getClass(), null);
+        final LocalizedMessage invalidPatternMessageMainInner = new LocalizedMessage(1,
+                "com.puppycrawl.tools.checkstyle.checks.naming.messages",
+                "name.invalidPattern", new String[] {"InputMainInner", "^[a-z0-9]*$"},
+                null, getClass(), null);
+        final String expectedPath = getFilePath("InputMain.java");
+        assertEquals(auditStartMessage.getMessage() + EOL
```
Comment (strkkk) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: EOL formatting can be changed to use AbstractPathTestSupport#addEndOfLine method to make it easier to read

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 5]
```diff
@@ -679,28 +670,25 @@ public void testExistingDirectoryWithViolations() throws Exception {
...
-            sb.append(auditFinishMessage.getMessage())
-                    .append(EOL);
-            assertEquals("Unexpected output log", sb.toString(), systemOut.getLog());
-            assertEquals("Unexpected system error log", "", systemErr.getLog());
-        });
-
         Main.main("-c", getPath("InputMainConfig-filelength.xml"),
                 getPath(""));
+        final String expectedPath = getFilePath("") + File.separator;
+        final StringBuilder sb = new StringBuilder(28);
```
Comment (strkkk) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: `StringBuilder(28);`
just wondering why it is like this...

File: src/test/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGeneratorTest.java (hunk 3 above)
Comment (pbludov) [reply src/test/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGeneratorTest.java]: > May be extract them to some method with `@BeforeEach ` annotation?

It works! Thank you very much.

File: src/test/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGeneratorTest.java (hunk 2 above)
Comment (pbludov) [reply src/test/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGeneratorTest.java]: Or even better
```
assertWithMessage("Unexpected output log")
    .that(systemOut.getCapturedData())
    .isEmpty();
```
But not in this PR. For this issue I would like the changeset to be as small as possible.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java (hunk 5 above)
Comment (pbludov) [reply src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: git blame says it comes from #4491. Well, any big enough value is good here.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java (hunk 4 above)
Comment (pbludov) [reply src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: done. This also had to be done separately, but they are few.

**Comment by rnveach** [comment]
@pbludov Blocked PR was merged. Please rebase the branch to pass CI.

**Comment by pbludov** [comment]
> @pbludov Blocked PR was merged. Please rebase the branch to pass CI.

done

**Pull request merged**